index.d.ts

```diff
@@ -5,103 +5,322 @@ import { GlobbyOptions } from 'globby';
+   * like template files.
+   */
   setPlopfilePath(filePath: string): void;
+  /**
+   * @returns the absolute path to the plopfile in use.
+   */
   getPlopfilePath(): string;
+  /**
+   * @returns the base path that is used when creating files.
+   */
   getDestBasePath(): string;
 
-  // plop.load functionality
+  /**
+   * Loads generators, helpers and/or partials from another plopfile or
+   * npm module.
+   */
   load(
     target: string[] | string,
     loadCfg: PlopCfg,
     includeOverride: boolean
   ): void;
+  /**
+   * Sets the default config that will be used for this plopfile if it is
+   * consumed by another plopfile using `plop.load()`.
+   */
   setDefaultInclude(inc: object): void;
+  /**
+   * Gets the default config that will be used for this plopfile if it is
+   * consumed by another plopfile using `plop.load()`.
+   */
   getDefaultInclude(): object;
 
+  /**
+   * Runs `template` through the handlebars template renderer using `data`.
+   * @returns the rendered template.
+   */
   renderString(template: string, data: any): String; //set to any matching handlebars declaration
 
   // passthroughs for backward compatibility
-  addPrompt(name: string, prompt: inquirer.PromptModule): void;
+  addPrompt: typeof inquirer.registerPrompt;
   addPartial(name: string, str: string): void;
   addHelper(name: string, fn: Function): void;
 }
 
 export interface PlopGenerator {
+  /**
+   * Short description of what this generator does.
+   */
   description: string;
+  /**
+   * Questions to ask the user.
+   */
   prompts: inquirer.Question[];
+  /**
+   * Actions to perform.
+   * If your list of actions needs to be dynamic, take a look at
+   * [using a dynamic actions array](https://plopjs.com/documentation/#using-a-dynamic-actions-array).
+   */
   actions: ActionType[];
 }
 
 export type CustomActionFunction = (
+  /**
+   * Answers to the generator prompts.
+   */
   answers: object,
+  /**
+   * The object in the `actions` array for the generator.
+   */
   config?: ActionConfig,
+  /**
+   * The plop api for the plopfile where this action is being run.
+   */
   plopfileApi?: NodePlopAPI
 ) => Promise<string> | string; // Check return type?
 
 export type ActionType =
   | ActionConfig
+  | AddActionConfig
   | AddManyActionConfig
   | ModifyActionConfig
   | AppendActionConfig
   | CustomActionFunction;
 
+/**
+ * There are several types of built-in actions you can use in your
+ * [GeneratorConfig](https://plopjs.com/documentation/#interface-generatorconfig).
+ * You specify which `type` of action (all paths are based on the location of
+ * the plopfile), and a template to use.
+ */
 export interface ActionConfig {
+  /**
+   * The type of action.
+   */
   type: string;
+  /**
+   * Overwrites files if they exist.
+   * @default false
+   */
   force: boolean;
-  data: object;
+  /**
+   *
+   * @default {}
+   */
+  data:
+    | object
+    | ((...args: any[]) => object)
+    | ((...args: any[]) => Promise<object>);
```

Comment:
Should this be changed to `((...args: any[]) => object | Promise<object>)`?

Reply:
See 74c372283de75feb75e43849dbdc582d76ddea19